packages/std/src/testing/assertions.rs

```diff
@@ -60,53 +60,57 @@ pub fn assert_approx_eq_impl<U: Into<Uint128>>(
     let rel_diff = Decimal::from_ratio(left.abs_diff(right), largest);
 
     if rel_diff > max_rel_diff {
-        match panic_msg {
-            Some(panic_msg) => panic!(
-                "assertion failed: `(left ≈ right)`\nleft: {left}\nright: {right}\nrelative difference: {rel_diff}\nmax allowed relative difference: {max_rel_diff}\n: {panic_msg}"
-            ),
-            None => panic!(
-                "assertion failed: `(left ≈ right)`\nleft: {left}\nright: {right}\nrelative difference: {rel_diff}\nmax allowed relative difference: {max_rel_diff}\n"
-            ),
-        }
+        do_panic(format_args!("assertion failed: `(left ≈ right)`\nleft: {left}\nright: {right}\nrelative difference: {rel_diff}\nmax allowed relative difference: {max_rel_diff}"), panic_msg);
     }
 }
 
-/// Tests that type `T` implements `Hash` trait correctly.
+/// Tests that type `T` implements `Eq` and `Hash` traits correctly.
 ///
 /// `left` and `right` must be unequal objects.
 ///
-/// Some object pairs may produce the same hash causing test failure.
-/// In those cases try different objects. The test uses stable hasher
-/// so once working pair is identified, the test’s going to continue
-/// passing.
+/// Some object pairs may produce the same hash causing test failure.  In those
+/// cases try different objects. The test uses stable hasher so once working
+/// pair is identified, the test’s going to continue passing.
 #[track_caller]
 #[doc(hidden)]
 #[cfg(test)]
-pub fn assert_hash_works_impl<T: Clone + Hash>(left: T, right: T, panic_msg: Option<String>) {
+pub fn assert_hash_works_impl<T: Clone + Eq + Hash>(left: T, right: T, panic_msg: Option<String>) {
```

Comment:
The `assert_hash_works` macro is exported, so this breaks any code using it on anything that doesn't implement `Eq` (not sure how likely that is, but still a breaking change).
Could you create a separate macro instead? Like `assert_hash_eq_works`?

Reply:
Actually nevermind. It wasn't released yet